wallet/wallet.go

```diff
@@ -557,7 +557,8 @@ func (w *Wallet) TSpendPolicy(tspendHash, ticketHash *chainhash.Hash) stake.Trea
 		if ok {
 			return policy
 		}
-	} else if policy, ok := w.tspendPolicy[*tspendHash]; ok {
+	}
+	if policy, ok := w.tspendPolicy[*tspendHash]; ok {
```

Comment:
Re-testing after the review fixes, I only have one test scenario where the result was unexpected but not necessarily wrong, so I'm going to point it out but will defer to what you or @jholdstock feels is most appropriate.

As currently implemented, the priority for deciding the voting policy is:

ticket + tspend policy
global tspend policy
ticket + key policy
global key policy

This means a vspd voting wallet configured with a global tspend policy (that is, a policy configured with `settspendpolicy <tspend> <yes|no>`) will override a ticket-specific key policy (one configured with `settreasurypolicy <pi key> <yes|no> <ticket>`).

Obviously, vspd voting wallets shouldn't have global policies set IMO (either per pi key or per tspend) so this could be a moot point anyway, but I wanted to surface it since it was unexpected to me.

Reply:
A vspd voting wallet should either use the per-ticket peference, or abstain. I dont think it should ever use a global policy.

vspd already does some checks on wallet config to ensure things like `enablevoting=true` and `unlocked=true`. If wallet were to support it, vspd could also perform checks to validate that no global policies are set.

We could even go as far as to have vspd remove any global policies from its voting wallets (again, if wallet were to support that).

Reply:
> [...] have vspd remove any global policies from its voting wallets (again, if wallet were to support that).

This is supported in this PR by calling `settreasurypolicy [key] abstain` or `settspendpolicy [tspend] abstain`. This removes the configured global policy.

> vspd could also perform checks to validate that no global policies are set.

I guess this could be marginally useful, though there's no guarantee the vsp operator won't just modify the global policy after vspd performs this check.

Ultimately, what needs to happen is for the wallet to be able to check the actual votes issued by a vsp and compare that to what the wallet configured the vsp to do so that we can it can build some stats on whether the vsp is actually voting according to user preferences and report that somewhere. Plenty of edge cases to deal with that though.

Reply:
So I think the important thing for now then is just to ensure that a per-ticket setting will override a global setting.